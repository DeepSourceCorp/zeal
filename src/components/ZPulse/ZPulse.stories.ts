import '../../assets/css/tailwind.css'
import '../../assets/css/typography.css'
import '../../assets/css/layout.css'
import ZPulse from './ZPulse.vue'

export default {
  title: 'Pulse Indicator',
  component: ZPulse,
  excludeStories: /.*Data$/
}

export const BasicPulse = () => ({
  components: { ZPulse },
  template: `<div>
      <z-pulse></z-pulse>
    </div>`
})

export const PulseWithColor = () => ({
  components: { ZPulse },
  template: `<div class="flex gap-2">
      <z-pulse color="cherry"></z-pulse>
      <z-pulse color="honey"></z-pulse>
      <z-pulse color="aqua"></z-pulse>
      <z-pulse color="slate"></z-pulse>
      <z-pulse color="robin"></z-pulse>
    </div>`
})

export const PulseWithSizes = () => ({
  components: { ZPulse },
  template: `<div class="flex flex-col gap-3">
      <div class="flex gap-2">
        <z-pulse size="small" color="cherry"></z-pulse>
        <z-pulse size="small" color="honey"></z-pulse>
        <z-pulse size="small" color="aqua"></z-pulse>
        <z-pulse size="small" color="slate"></z-pulse>
        <z-pulse size="small" color="robin"></z-pulse>
      </div>
      <div class="flex gap-2">
        <z-pulse color="cherry"></z-pulse>
        <z-pulse color="honey"></z-pulse>
        <z-pulse color="aqua"></z-pulse>
        <z-pulse color="slate"></z-pulse>
        <z-pulse color="robin"></z-pulse>
      </div>
      <div class="flex gap-2">
        <z-pulse size="large" color="cherry"></z-pulse>
        <z-pulse size="large" color="honey"></z-pulse>
        <z-pulse size="large" color="aqua"></z-pulse>
        <z-pulse size="large" color="slate"></z-pulse>
        <z-pulse size="large" color="robin"></z-pulse>
      </div>
    </div>`
})
