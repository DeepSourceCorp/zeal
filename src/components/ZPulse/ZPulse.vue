<template>
  <div :class="`flex flex-shrink-0 h-${pulseSize} w-${pulseSize} items-center justify-center`">
    <span
      :class="`animate-ping absolute inline-flex h-${pulseSize} w-${pulseSize} rounded-full border-2 border-${color} bg-opacity-75`"
    ></span>
    <span :class="`relative inline-flex rounded-full h-${dotSize} w-${dotSize} bg-${color}`"></span>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'

const SIZEMAP = {
  small: { pulse: 2, dot: 1 },
  base: { pulse: 3, dot: 2 },
  large: { pulse: 5, dot: 3 }
}

export default Vue.extend({
  name: 'ZPulse',
  props: {
    color: {
      type: String,
      default: 'juniper'
    },
    size: {
      type: String,
      default: 'base',
      validator: function (value: string): boolean {
        return ['small', 'base', 'large'].includes(value)
      }
    }
  },
  computed: {
    pulseSize(): string {
      return SIZEMAP[this.size || 'base'].pulse
    },
    dotSize(): string {
      return SIZEMAP[this.size || 'base'].dot
    }
  }
})
</script>
