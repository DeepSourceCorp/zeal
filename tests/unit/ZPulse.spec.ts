import ZPulse from '../../src/components/ZPulse'

import { shallowMount } from '@vue/test-utils'

const template = `<div class="flex flex-col gap-3">
<div class="flex gap-2">
  <z-pulse size="small" color="cherry"></z-pulse>
  <z-pulse size="small" color="honey"></z-pulse>
  <z-pulse size="small" color="aqua"></z-pulse>
  <z-pulse size="small" color="slate"></z-pulse>
  <z-pulse size="small" color="robin"></z-pulse>
</div>
<div class="flex gap-2">
  <z-pulse color="cherry"></z-pulse>
  <z-pulse color="honey"></z-pulse>
  <z-pulse color="aqua"></z-pulse>
  <z-pulse color="slate"></z-pulse>
  <z-pulse color="robin"></z-pulse>
</div>
<div class="flex gap-2">
  <z-pulse :active="false" color="cherry"></z-pulse>
  <z-pulse :active="false" color="honey"></z-pulse>
  <z-pulse :active="false" color="aqua"></z-pulse>
  <z-pulse :active="false" color="slate"></z-pulse>
  <z-pulse :active="false" color="robin"></z-pulse>
</div>
<div class="flex gap-2">
  <z-pulse size="large" color="cherry"></z-pulse>
  <z-pulse size="large" color="honey"></z-pulse>
  <z-pulse size="large" color="aqua"></z-pulse>
  <z-pulse size="large" color="slate"></z-pulse>
  <z-pulse size="large" color="robin"></z-pulse>
</div>
</div>`

describe('Pulse', () => {
  it('default rendering', () => {
    const wrapper = shallowMount(ZPulse)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders with options', () => {
    const wrapper = shallowMount(ZPulse, {
      template: template
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
