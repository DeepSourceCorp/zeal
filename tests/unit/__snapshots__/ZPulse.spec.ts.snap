// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pulse default rendering 1`] = `<div class="flex flex-shrink-0 h-4 w-4 items-center justify-center"><span class="absolute inline-flex rounded-full h-3 w-3 bg-juniper animate-ping"></span> <span class="relative inline-flex rounded-full h-3 w-3 bg-juniper"></span></div>`;

exports[`Pulse renders with options 1`] = `<div class="flex flex-shrink-0 h-4 w-4 items-center justify-center"><span class="absolute inline-flex rounded-full h-3 w-3 bg-juniper animate-ping"></span> <span class="relative inline-flex rounded-full h-3 w-3 bg-juniper"></span></div>`;
